pkgs/tools/security/ghidra/lib/ghidra.nix

```diff
@@ -0,0 +1,67 @@
+#TODO It may be possible to use symlinkjoin to add plugins if you write the unwrapped launcher script to not use readlink
+
+# plugins is a list of derivations to symlink into <ghidraroot>/Ghidra/Extensions,
+# this is entirely sufficient for ghidra to operate.
+{
+  stdenv, fetchurl, unzip, makeWrapper, autoPatchelfHook, # Utilities
+  jdk, pam, # Deps
+  lib, config, plugins ? [], extraLaunchers ? {}, # Ghidra
+  pkgs #.lib
+  }:
+
+#Recursive let means these two need to be separate
+let
+  inherit (lib) jdkWrapper installPlugin unpackPlugin writeCustomLauncher;
+in
+let
+  lib = pkgs.lib;
+in
+
+  stdenv.mkDerivation {
+    name = "ghidra-9.0";
+
+    src = fetchurl {
+      url = https://ghidra-sre.org/ghidra_9.0_PUBLIC_20190228.zip;
+      sha256 = "3b65d29024b9decdbb1148b12fe87bcb7f3a6a56ff38475f5dc9dd1cfc7fd6b2";
+      };
+
+    nativeBuildInputs = [
+      makeWrapper
+      autoPatchelfHook
+      unzip
+      ];
+
+    # For autoPatchelf
+    buildInputs = [
+      stdenv.cc.cc.lib
+      pam
+      ];
+
+    propagatedBuildInputs = [
+      jdk
```

Comment:
Should jdk really be propagated here?

Reply:
Shouldn't it? I think I wrote this when I wasn't so good at nix.
I guess at the time this was providing me with the setup hook, which I now call manually in jdkWrapper, making this unnecessary?

Reply:
Yes. I don't see why derivations that receive ghidra as build input also need access to the same jdk binary. The wrapper ensures that java can be found at runtime.